### ISSUE TITLE
debug.traceback of loadstring get unknown file

### ISSUE THREAD

**Issue by pilaoda** [TypeScriptToLua/TypeScriptToLua#1259]
My traceback:
```
Script Runtime Error: unknown:743: attempt to perform arithmetic on local 'c' (a string value)
stack traceback:
    unknown:743: in function 'handler'
    unknown:189: in function <unknown:186>
```

My code:
```
local encryptedCode = "XXXXXX"
local decryptedCode = decrypt(encryptedCode)
loadstring(decryptedCode , "sys/Cheat.lua")()
```

What I have done is encrypt the generated lua file, and use `loadstring` after decryption.
I already set the file path as the second parameter `chunkname` of loadstring, but the traceback still be replaced by this code: https://github.com/TypeScriptToLua/TypeScriptToLua/blob/288ab8cd1c7e1736578db0b31fd23f62c0ccfce6/src/lualib/SourceMapTraceBack.ts#L46

What I expect is using the sourcemap with the `chunkname` I set in loadstring.
I have two ideas to achieve this:
1.  Assume the chunkname is the lua path, and replace the traceback substring `[string "xx/xx.lua"]:22` to `xx/xx.ts:10` or `[string "xx/xx.ts"]:10`
2.  Provide a **custom replacer** of `loadstring` for users to implement by themselves.

**Comment by pilaoda** [comment]
If my idea is accepted, I would like to write a PR for this.

**Comment by Perryvw** [comment]
Option 1 sounds fine to me, I expected that to work already, PR to correct the behaviour is welcome.

**Comment by pilaoda** [comment]
PR to fix this with Option1: #1260
